src/libraries/System.IO.FileSystem/tests/FileStream/ctor_str_fm_fa_fs_buffer_fo.cs

```diff
@@ -67,6 +66,47 @@ public void ValidFileOptions(FileOptions option)
             }
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [InlineData((FileOptions)0x20000000)] // FILE_FLAG_NO_BUFFERING on Windows
+        [InlineData(FileOptions.Asynchronous | (FileOptions)0x20000000)]
+        public unsafe void NoBufferingOptions(FileOptions option)
+        {
+            void* pWrite = NativeMemory.AlignedAlloc(c_DefaultBufferSize, c_DefaultBufferSize);
+            void* pRead = NativeMemory.AlignedAlloc(c_DefaultBufferSize, c_DefaultBufferSize);
```

Comment:
Why two buffers rather than one?

Reply:
According to the man page:
```
       the alignment of the user
       buffer, and the file offset must all be multiples of the logical
       block size of the filesystem
```
How are you guaranteeing that this 4096 alignment is valid everywhere this test runs?